pkg/chunk/storage/factory.go

```diff
@@ -69,3 +71,25 @@ func NewTableClient(cfg Config) (chunk.TableClient, error) {
 		return nil, fmt.Errorf("Unrecognized storage client %v, choose one of: aws, gcp, inmemory", cfg.StorageClient)
 	}
 }
+
+// Opts returns the StorageOpts for the given configuration.
+func Opts(cfg Config, schemaCfg chunk.SchemaConfig) ([]chunk.StorageOpt, error) {
+	opts := []chunk.StorageOpt{}
+	client, err := NewStorageClient(cfg, schemaCfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating storage client")
+	}
+
+	opts = append(opts, chunk.StorageOpt{From: model.Time(0), Client: client})
+	if schemaCfg.BigtableColumnKeyFrom.IsSet() {
+		cfg.GCPStorageConfig.ColumnKey = true
```

Comment:
Can we avoid copy config like this?  Can you make UseColumnKey a param on the bigtable constructor, and have NewStorageClient look in schemaCfg?

Reply:
Yes, but in both cases the `schemaConfig` will be the same. We need to return two stores, one with and one without if the option is set.

I've made a better modification but that looks like bleeding bigtable changes into other things too....